src/main/java/com/example/onlinebookstore/model/OrderItem.java

```diff
@@ -0,0 +1,41 @@
+package com.example.onlinebookstore.model;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.ManyToOne;
+import jakarta.persistence.Table;
+import java.math.BigDecimal;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.Where;
+
+@Entity
+@Data
+@SQLDelete(sql = "UPDATE order_item SET is_deleted = true WHERE id=?")
+@Where(clause = "is_deleted=false")
+@Table(name = "order_item")
+public class OrderItem {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @ManyToOne
+    @JoinColumn(name = "order_id", nullable = false)
+    private Order order;
```

Comment:
I recommend to exclude all relations from equals, hashcode and toString to prevent unexpected problems in the future

Reply:
not resolved

Reply:
not resolved